order-book/src/main/java/com/market/banica/order/book/service/grpc/OrderBookService.java

```diff
@@ -37,24 +34,28 @@ private OrderBookService(final AuroraClient auroraClient, final ItemMarket itemM
     }
 
     @Override
-    public void getOrderBookItemLayers(ItemOrderBookRequest request, StreamObserver<ItemOrderBookResponse> responseObserver) {
-
-        Optional<Set<Item>> result = itemMarket.getItemSetByName(request.getItemName());
-
-        if (result.isPresent()) {
-            responseObserver.onNext(ItemOrderBookResponse.newBuilder()
-                    .setItemName(request.getItemName())
-                    .addAllOrderbookLayers(result.get().stream()
-                            .map(item -> OrderBookLayer.newBuilder()
-                                    .setPrice(item.getPrice())
-                                    .setQuantity(item.getQuantity())
-                                    .setOrigin(item.getOrigin())
-                                    .build()).collect(Collectors.toList()))
-                    .build());
+    public void getOrderBookItemLayers(Aurora.AuroraRequest request, StreamObserver<Aurora.AuroraResponse> responseObserver) {
+        String[] topicSplit = request.getTopic().split(REGEX);
+        String itemName = topicSplit[1];
+        long itemQuantity = Long.parseLong(topicSplit[2]);
+
+        List<OrderBookLayer> requestedItem = itemMarket.getRequestedItem(itemName, itemQuantity);
+
+        if (requestedItem.size() > 0) {
+            responseObserver.onNext(
```

Comment:
both if and else are doing absolutely the same thing, it could be extracted as a method and to pass ItemOrderBookResponse as a parameter

Reply:
Fixed